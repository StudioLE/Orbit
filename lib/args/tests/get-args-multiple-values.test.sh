#!/bin/bash
set -uo pipefail

# ARGS
ACTUAL_FILE=$1

# SETUP

source /srv/lib/args/get-args.sh

# START

get-args \
  --string "Hello, world!" \
  --int 1 \
  --flag \
  --optional value \
  -s "Short, value" \
  -a \
  -bc \
  -1 "Short int" \
  -23 \
  -- \
  /path/to/file \
  path/to/file

declare OUTPUT

OUTPUT=""

for key in "${!ARGS[@]}"
do
  OUTPUT+="$key: ${ARGS[$key]}"
  OUTPUT+=$'\n'
done

OUTPUT=$(echo "${OUTPUT}" | sed '$d')

echo "${OUTPUT}" > "${ACTUAL_FILE}"
