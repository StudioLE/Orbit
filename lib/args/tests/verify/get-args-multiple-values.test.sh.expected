--int: 1
--string: Hello, world!
1: path/to/file
0: /path/to/file
--optional: value
--flag: TRUE
